### PR TITLE
D.js file for duckduckgo for issue #195

### PR DESCRIPTION
**Pull request by vikas-kundu**
Here is a gif: https://asciinema.org/a/J4fTuza9sY4Bq2ooVQt1IKBy7
Also, is this output format ok? Or how should it be?

### CODE REVIEW

**Comment by saeeddhqan** [comment]
if you can along with this format, extract data like what google module does

**Comment by vikas-kundu** [comment]
Working on it

**Comment by vikas-kundu** [comment]
- Instead of Google Page-speed API, using direct source duckduckgo as d.js is present on the page itself.
- Improved output format into JSON
I have the clean output in JSON format, should I send only the values of dict items into the output ?

File: modules/search/d_js.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	d_js_url = search(self, query)
+	if not d_js_url:
+		return output
+
+	self.verbose('[d_js] Parameters for d.js file found, sending direct request...')
+	try:
+		# Redirects should be False to avoid error redirects after many tries.
+		d_js_data = self.request(url=d_js_url, allow_redirects=False, headers=header_data).text		
+	except Exception as e:
+		self.error('Unable to reach duckduckgo', 'modules/search', 'module_api')
```
Comment (saeeddhqan) [inline modules/search/d_js.py]: d_js instead of 'modules/search'

**Review by saeeddhqan — COMMENTED** [review]
the output should be exactly like google, yahoo or dogpile. `[{'title': '...', 'a': 'link', 'cite': '...', 'content': 'desccription'}, ...]`

File: modules/search/d_js.py (hunk 1 above)
Comment (vikas-kundu) [reply modules/search/d_js.py]: Made the necessary changes. Is the cite url ok?

**Comment by saeeddhqan** [comment]
Could I see a picture from the response?

**Comment by saeeddhqan** [comment]
Tested. great!

Thank you.

**Pull request merged**